### ISSUE TITLE
[rewrite] oncreate not firing in strange edge case

### ISSUE THREAD

**Issue by mindeavor** [lhorie/mithril.js#1163]
### Description:

When your DOM tree structure looks a certain way, an `oncreate` callback does not fire.

### Steps to Reproduce:

JS Bin: http://jsbin.com/fevifa/1/edit?js,output

```js
var flag = false;

var Example  = {
  view : function( component ){
    return m('.example-component', [
      flag
        ? m('.graph', { oncreate: () => alert("oncreate!") })
        : m('p', "Loading...")
      ,
      m('.one'),
      m('.two'),
    ]);
  }
};

m.mount( document.body, {
  view: () => m('.app', [
    m('button', { onclick: () => flag = !flag }, "Toggle"),
    m(Example, {}),
  ])
});
```

### Expected:

An alert should pop up when clicking the `toggle` button.

### Actual:

Nothing happens. However, if you remove `m('.one')` or `m('.two')`, everything works as expected.

**Comment by lhorie** [comment]
Fixed in 1155027c9125f9d157318d0c2d7583b90e2e1832

**Comment by mindeavor** [comment]
🎉
